### PR TITLE
executor: add cast for admin check table/index to avoid out of range

### PR DESCRIPTION
**Pull request by wjhuang2016**
### What problem does this PR solve?

Issue Number: ref #46176

Problem Summary:
Add cast function to avoid out of range.

### Check List

Tests 

- [x] Unit test
- [x] Integration test
Doesn't out of range in an internal CI test.
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: executor/executor.go [hunk 1]
```diff
@@ -2457,8 +2457,8 @@ func (w *checkIndexWorker) HandleTask(task checkIndexTask) {
 			logutil.BgLogger().Warn("compare checksum by group reaches time limit", zap.Int("times", times))
 			break
 		}
-		whereKey := fmt.Sprintf("((%s - %d) %% %d)", md5Handle.String(), offset, mod)
-		groupByKey := fmt.Sprintf("((%s - %d) div %d %% %d)", md5Handle.String(), offset, mod, bucketSize)
+		whereKey := fmt.Sprintf("((cast(%s as signed) - %d) %% %d)", md5Handle.String(), offset, mod)
+		groupByKey := fmt.Sprintf("((cast(%s as signed) - %d) div %d %% %d)", md5Handle.String(), offset, mod, bucketSize)
```
Comment (tangenta) [inline executor/executor.go:2461]: Could you add a functional test?

**Review by tangenta — APPROVED** [review]

File: executor/executor.go [hunk 2]
```diff
@@ -2457,8 +2457,8 @@ func (w *checkIndexWorker) HandleTask(task checkIndexTask) {
 			logutil.BgLogger().Warn("compare checksum by group reaches time limit", zap.Int("times", times))
 			break
 		}
-		whereKey := fmt.Sprintf("((%s - %d) %% %d)", md5Handle.String(), offset, mod)
-		groupByKey := fmt.Sprintf("((%s - %d) div %d %% %d)", md5Handle.String(), offset, mod, bucketSize)
+		whereKey := fmt.Sprintf("((cast(%s as signed) - %d) %% %d)", md5Handle.String(), offset, mod)
```
Comment (zimulala) [inline executor/executor.go:2460]: Why do we add "cast(...as signed) " can fix `BIGINT value is out of range`?

**Review by zimulala — APPROVED** [review]
LGTM

**Pull request merged**